services/storage/predicate_influxql.go

```diff
@@ -242,6 +242,56 @@ func (v *hasRefs) Visit(node influxql.Node) influxql.Visitor {
 	return v
 }
 
+// HasSingleMeasurementNoOR determines if an index optimisation is available.
+//
+// Typically the read service will use the query engine to retrieve all field
+// keys for all measurements that match the expression, which can be very
+// inefficient if it can be proved that only one measurement matches the expression.
+//
+// This condition is determined when the following is true:
+//
+//		* there is only one occurrence of the tag key `_measurement`.
+//		* there are no OR operators in the expression tree.
```

Comment:
If there's only one occurrence of `_measurement`, does it matter if there is an OR operator?

Reply:
@dgnorton yes because you can have `_measurement = 'foo' OR _measurement = 'bar'`

Reply:
That doesn't count as one measurement?

Reply:
No. In this model, a measurement is just a tag called `_measurement`, therefore we can't tell easily how many measurements in the index we need to select from.

Reply:
@dgnorton flux would allow a query like:

```flux
filter(fn: (d) => (d._measurement == "cpu" and d._field == "idle") OR (d._measurement == "mem" and d._field == "free")
```

we would like to add future improvements, such that the optimization would still apply in the following scenario, which still applies to a single measurement:

```flux
filter(fn: (d) => (d._measurement == "cpu" and (d._field == "system" OR d._field == "user"))
```